pyhap/characteristic.py

```diff
@@ -184,7 +187,11 @@ def to_valid_value(self, value):
                 logger.error(error_msg)
                 raise ValueError(error_msg)
         elif self.properties[PROP_FORMAT] == HAP_FORMAT_STRING:
-            value = str(value)[:256]
+            max_length = min(
+                ABSOLUTE_MAX_LENGTH,
```

Comment:
It might make more sense to throw a value error in `override_properties` if they try to set it to something larger than `ABSOLUTE_MAX_LENGTH`